spec/lexers/syzprog_spec.rb

```diff
@@ -0,0 +1,136 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+describe Rouge::Lexers::Syzprog do
+  let(:subject) { Rouge::Lexers::Syzprog.new }
+
+  describe 'lexing' do
+    include Support::Lexing
+
+    it 'recognizes syscalls' do
+      assert_tokens_equal "foo$bar(&AUTO=@field, &(0x100)={0x0, nil, @void})",
```

Comment:
**suggestion:** We tend not to write specs for lexer this way since asserting to specific tokens make the tests very brittle. Could we add these samples in `spec/visual/samples/syzprog` and verify via the visual tester (rackup and visit `http://localhost:9292`)

Reply:
Hm, but then to verify the correctness I would need to do it manually by looking at the result? This is what seems brittle :)

The tests are there to catch breakages. If a lexer change causes the tests to break due to the tests being imprecise in the first place, they are updated with an explicit justification. Otherwise, there's a bug in the lexer change. And this bug was automatically detected, which is what the tests are for.

I would keep the tests as is, if you don't mind.

Reply:
No worries @xairy. Thank you for expanding on your thoughts. I don't have a strong opinion against it anyway. Happy to leave them as is 👍🏼